bot.js

```diff
@@ -10,22 +10,53 @@ client.once('ready', () => {
 client.login(auth.token);
 
 client.on('message', message => {
-	console.log(message.content);
-  if (!message.content.includes("&") || message.author.bot) return;
+  if(message.author.bot) return;
 
-  const args = message.content.slice(message.content.search("&")+1).split(' ');
-  const command = args.shift().toLowerCase();
+  console.log(message.content);
+  
+  var mention = false;
+  if(message.mentions.users.size > 0)
+  {
+    //check if mention contains app client id
+    if(message.mentions.users.first().id == auth.id)
+    {
+      mention = true;
+    }
+  }
+
+  if (!message.content.includes("&") && !mention) return;
+
+  var args = message.content.slice(message.content.search("&")+1).split(' ');
+  var command = args.shift().toLowerCase();
+  if(mention)
+  {
+    command = 'deyoh';
+  }
 
   if (command === 'ping') {
 	  // send back "Pong." to the channel the message was sent in
 	  message.channel.send('Pong.');
   }
   else if (command === 'deyohtime' || command === 'deyoh') {
-    if (args[0] == 'now') {
-       message.channel.send(moment().tz("America/Punta_Arenas").format("h:mmA"))
+    var timezone = "America/Punta_Arenas";
+    if(args[1] != null)
+    {
+      zoneOption = args[1];
+      zones = moment.tz.names();
+      zones.forEach(name => {
+        if(name.toLowerCase().includes(zoneOption.toLowerCase()))
+        {
+          zoneOption = name;
+        }
+      });
+      timezone = zoneOption;
```

Comment:
so, i think in the case that it does not find a matching timezone, it will reset the timezone variable to whatever the 2nd word after the command is, which is less than desirable, especially if a mid sentence usage of the bot

Reply:
My bad on that! Now uses default timezone if nothing is matched on.